creator-node/src/index.ts

```diff
@@ -122,10 +122,17 @@ const startApp = async () => {
   serviceRegistry.initServicesThatRequireServer(appInfo.app)
 
   // when app terminates, close down any open DB connections gracefully
-  ON_DEATH((signal: any, error: any) => {
+  ON_DEATH(async (signal: any, error: any) => {
```

Comment:
this should work. an async fn is just a fn that returns a promise. you should be able to pass it in.

you may run into some weirdness w/ the specific signal that's terminating the process and whether all of the async stuff does truly get run https://github.com/jprichardson/node-death/issues/7#issuecomment-421505954

but i think if this is best effort, that's fine/as good as it can get for termination signal handling. a better assurance might be to have the audius-cli bit in the compose set up first drain traffic and then send a separate wrap up jobs command, then then shut down/upgrade